### PR TITLE
Validation DS100

### PR DESCRIPTION
**Pull request by tkatzir**
## Related Issues
relates: https://jira-dc.paloaltonetworks.com/browse/CIAC-7792

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/DS_validators/DS100_description_missing_in_beta_integration.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+)
+
+ContentTypes = Integration
+
+
+class DescriptionMissingInBetaIntegrationValidator(BaseValidator[ContentTypes]):
+    error_code = "DS100"
+    description = "Check whether a description file exists for a beta integration."
+    rationale = "Need a disclaimer for beta integrations."
+    error_message = "Beta integration needs a description"
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/DS_validators/DS100_description_missing_in_beta_integration.py:21]:
```suggestion
    error_message = "Beta integration needs a description."
```

File: demisto_sdk/commands/validate/validators/DS_validators/DS100_description_missing_in_beta_integration.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        return [
+            ValidationResult(
+                validator=self,
+                message=self.error_message,
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if content_item.description_file.exist
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/DS_validators/DS100_description_missing_in_beta_integration.py:38]:
```suggestion
            if content_item.is_beta and not content_item.description_file.exist
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Looks good!
Make sure to add tests as well.

File: .changelog/4274.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Refactored validation DS100, check whether a description file exists for a beta integration.
+  type: feature
+pr_number: 4269
```
Comment (DeanArbel) [inline .changelog/4274.yml:4]:
```suggestion
pr_number: 4274
```

**Review by DeanArbel — APPROVED** [review]
looks good, fix error

File: demisto_sdk/commands/validate/tests/DS_validators_test.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    Given
+    content_items iterables.
+    When
+    - Calling the DescriptionMissingInBetaIntegrationValidator is valid function.
+    Then
+        - Make sure that the validation is implemented correctly for beta integrations.
+        - Case 1: Not a beta integration, and no description file.
+        - Case 1: Not a beta integration, with a description file.
+        - Case 1: A beta integration, and no description file.
+        - Case 1: A beta integration, with a description file.
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/DS_validators_test.py:48]:
```suggestion
    Given
    content_items iterables.
            - Case 1: Not a beta integration, and no description file.
            - Case 2: Not a beta integration, with a description file.
            - Case 3: A beta integration, and no description file.
            - Case 4: A beta integration, with a description file.
    When
    - Calling the DescriptionMissingInBetaIntegrationValidator is valid function.
    Then
        - Make sure that the validation is implemented correctly for beta integrations.
        - Case 1: Shouldn't fail.
        - Case 2: Shouldn't fail.
        - Case 3: Should fail.
        - Case 4: Shouldn't fail.
```

File: demisto_sdk/commands/validate/validators/DS_validators/DS100_description_missing_in_beta_integration.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+ContentTypes = Integration
+
+
+class DescriptionMissingInBetaIntegrationValidator(BaseValidator[ContentTypes]):
+    error_code = "DS100"
+    description = "Check whether a description file exists for a beta integration."
+    rationale = "Need a disclaimer for beta integrations."
+    error_message = "Beta integration needs a description."
+    related_field = ""
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/DS_validators/DS100_description_missing_in_beta_integration.py:22]:
```suggestion
    related_field = "beta"
```

**Review by YuvHayun — APPROVED** [review]
Minor comments.

**Pull request merged**